akka-stream/src/main/scala/akka/stream/impl/SubFlowImpl.scala

```diff
@@ -34,6 +34,8 @@ class SubFlowImpl[In, Out, Mat, F[+_], C](val subFlow: Flow[In, Out, NotUsed],
   override def named(name: String): SubFlow[Out, Mat, F, C] =
     new SubFlowImpl[In, Out, Mat, F, C](subFlow.named(name), mergeBackFunction, finishFunction)
 
+  override def async: Repr[Out] = new SubFlowImpl[In, Out, Mat, F, C](subFlow.async, mergeBackFunction, finishFunction)
```

Comment:
I started out like that but the compiler was complaining about it when I put parens in the Java DSL. But I guess I can just omit the parens there as well.